### DISCUSSION TITLE
Go back to where bookmark was taken

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2954] [Ideas]
### Idea Description

Right now there isn't a way to jump back into the series where the bookmark was taken from. I've marked some really awesome pages as bookmarks but I don't know what volume / chapter or even what page number it was from. Having a link back to that page in the source would be awesome for going back and re-reading the segment again.

### Idea Category

User Experience

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I'd actually like to do a whole overhaul on the bookmark system and would love some ideas around it. Especially around handling bookmarks when the source series/chapter/volumes are removed from Kavita (temp or permanently).

**Reply by DieselTech** [reply]
One of the things I was thinking would be nice is to have a per series view of the bookmarks. That way when you go into the bookmarks for the series you can see each page you've saved. That might give you some room to display a little bit of extra info around the page as well. That way it's not all in the image reader.

**Reply by Teagana-999** [reply]
I see this is a few years old but:

The current bookmarks, as I understand them, feel more like a gallery.  Which makes sense as a feature for books with art, but isn't what I think when I think bookmark.  Perhaps that option should be renamed "add page to gallery" or something.

For bookmarks, I want to be able to save bookmarks within a book, with a page number and probably a simple name.  Ideally, in two clicks, one for the bookmark button, one to confirm the name.  Then, I want to be able to do something like open a side panel while I'm reading, select a bookmark I've saved, and be taken directly to that page.  I want a navigation aid.  Especially for reference books.

**Reply by majora2007** [reply]
There are 2 types of bookmarks @Teagana-999 , there are image-based ones and text-based and they work differently. Image-based are what you're familiar with, they are more of a gallery. Text-based are more like a bookmark in a book, a way to jump to a fav cookie recipe. 

The epub reader can create both (and in this case, the epub menu can jump to either). So if you're reading epubs, Kavita already has this. If you're reading comic books, it's not yet implemented. The code is there, I just never got back around to updating the reader to allow for linking in a nice UX.

**Reply by Teagana-999** [reply]
Ah, that's good to know.  I have some epubs, but also a few pdfs, I couldn't find bookmark functionality in my pdfs.

**Reply by majora2007** [reply]
Unfortunately the pdf reader is extremely limited in what can be done and it's not a primary focus for us right now. You can raise a FR for explicit PDF support for Bookmarks (Bookmark Parity for PDF). I would need to see a lot of support from the community to be able to take that on. 

I def want better parity with our PDF reader, but there is only so much Amelia and I can do.

**Reply by Teagana-999** [reply]
Fair.

**Comment by Menghini** [comment]
How are bookmarks or highlights handled in commercial apps along with read progress? Do you think they are saved x words into a document/chapter? With epubs not having pages, but rather separated into chapters, I think it would be more beneficial to keep track of these by seeing how far you are along in a chapter.

I do think it is hard to visualize how to handle file changes. But at the same time, how often is a epub file going to change? In theory, these are published and final?

**Reply by majora2007** [reply]
You're talking about something unrelated to this Feature Request. Bookmarks are a image-based functionality only. Epubs have what are called Personal Table of Contents to offer an ability to jump back to certain places. 

Whereas, bookmarks are images that are saved externally for quick viewing.

**Reply by Menghini** [reply]
I apologize for my error. I was able to "bookmark" (I guess this is my personal Table of Contents?) within an epub. I guess the language is confusing with this? I did read a comment somewhere along the way of why this doesn't show up in the "Bookmarks" tab of the dashboard of Kavita.
![image](https://github.com/user-attachments/assets/57a6b470-e78a-4809-bbc0-d7398a856cca)

**Reply by majora2007** [reply]
Yeah it's a bit of a poor naming on my part. Personal ToC was too long for the space available and that's also why it can't show up on the Bookmarks, because when I created it, it's mainly for just images you want to save for later, whereas PToC is for places you need to save, like favorite recipes, etc.

**Reply by FlyAg** [reply]
Hello, I know you were asking about recommendations to this system and I was thinking since it is a bit of a poor naming and I got confused as well, if these so called PToC could be tabulated into the existing Bookmarks page, showing the little thumbnail of the eBook cover and perhaps when clicked upon, takes you to either the viewer with the PToC Bookmarks opened up or to a page that the bookmarks listed out that you can click on to go to the viewer.
